ucm2/HDA/HiFi-mic.conf

```diff
@@ -0,0 +1,451 @@
+# Generic handling of analog microphones on HDA devices
+
+# Microphone related variables on entry
+# If DMIC is present:
+#   (DeviceDmic == "Mic1")
+#   DeviceMic == "Mic2"
+#
+# If DMIC is not present:
+#   DeviceMic == "Mic"
+
+# Microphone related variables as used internally
+# HDA codecs can offer input source selection (mux) via "Input Source" or
+# "Capture Source" with varying options:
+#  'Internal Mic'
+#  'Headset Mic'
+#  'Headphone Mic'
+#  'Mic'
+#
+# or there can be 'Mic' and optional 'Front Mic' available
+#
+# This leads to up to maximum of four possible input devices.
+
+Define.SourceControl ""
+Define.MicJackControl ""
+
+# DeviceMic is pre-configured
+Define.DeviceMicName ""
+Define.DeviceMicComment ""
+Define.DeviceMicPriority ""
+Define.DeviceMicJack ""
+Define.DeviceMic2 ""
+Define.DeviceMic2Name ""
+Define.DeviceMic2Comment ""
+Define.DeviceMic2Priority ""
+Define.DeviceMic2Jack ""
+Define.DeviceMic3 ""
+Define.DeviceMic3Name ""
+Define.DeviceMic3Comment ""
+Define.DeviceMic3Priority ""
+Define.DeviceMic3Jack ""
+Define.DeviceMic4 ""
+Define.DeviceMic4Name ""
+Define.DeviceMic4Comment ""
+Define.DeviceMic4Priority ""
+Define.DeviceMic4Jack ""
+
+# evaluate the microphone jack name
+If.micjackctl {
+	Condition {
+		Type ControlExists
+		Control "iface=CARD,name='Headset Mic Phantom Jack'"
```

Comment:
I would not use the "software defined" Phantom Jacks at all. They were defined for pulseaudio, because their port selection algorithm was designed in a simple way. The conflicting device links are resolving this issue. This block can just use other controls to do similar job.

I think that this decision tree should fit:

1. Headset Mic Jack
2. Headphone Mic Jack
3. Headphone Jack
4. Mic Jack